params.go

```diff
@@ -377,6 +377,25 @@ func BuildQueryString(opts interface{}) (*url.URL, error) {
 	return nil, fmt.Errorf("Options type is not a struct.")
 }
 
+/*
+BuildQueryStringWithFilters is an internal function to be used by request methods in
+individual resource packages.
+
+It accepts a tagged structure and a map and expands them into a URL struct.
+*/
+func BuildQueryStringWithFilters(opts interface{}, filters map[string]string) (*url.URL, error) {
+	u, err := BuildQueryString(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	params := u.Query()
+	for k, v := range filters {
+		params.Add(k, v)
+	}
+	return &url.URL{RawQuery: params.Encode()}, nil
```

Comment:
I realize this function will reduce the amount of code in the identity package, but since it's not used outside of identity, it's better to modify _each_ `ToXListQuery` function.

This is already done in a handful of other Gophercloud packages. 

See here for a simple example: https://github.com/gophercloud/gophercloud/blob/master/openstack/clustering/v1/webhooks/requests.go#L22-L32